### PR TITLE
Add static analysis tools to the CI build #5267

### LINKED ISSUE #5267: Add static analysis tools to the CI build
As per title. I've worked on this for quite some time but (believe it or not) it's put on hold by #5255. Time to finally surface it.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5267

### CODE REVIEW

**Comment by wkurniawan07** [comment]
As mentioned, I've actually worked on this for some time so I'm considerably familiar with all the tools added. Few pointers to take note of:

1. The successive failing builds above are on purpose; they're showing the static analysis tools in action, although some didn't show what they're supposed to show because I pushed the subsequent commit too soon :/
2. The static analysis is done _before_ testing, so that we don't have to waste time and resources running tests when the code is not standard-compliant (running the tools locally is also very easy).
3. All the tools used here are supported in Eclipse, with the exception of ESLint. More details can be found in [the documentation I provided](https://github.com/TEAMMATES/teammates/blob/5267-add-static-analysis-tools/devdocs/staticAnalysis.md).
4. We have complete control over the ruleset of all tools. This is how I can minimize the ruleset used right now, because I don't want to open 4 cans of worms just yet.

No longer will we have to worry about whitespace issues etc. now (soon)! 🎵🎵

Feel free to ask me if any clarification is needed.

File: devdocs/staticAnalysis.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+
```
Comment (damithc) [inline devdocs/staticAnalysis.md]: This file should be linked from README.md and other relevant places?

File: devdocs/staticAnalysis.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# Static Analysis
+
+TEAMMATES uses a number of static analysis tools in order to maintain code quality and measure code coverage.
+This document will cover an overview of these tools and how to run them in local environment.
+
+- [Tool stack](#tool-stack)
+- [Running static analysis](#running-static-analysis)
+- [Running code coverage session](#running-code-coverage-session)
+
+## Tool stack
```
Comment (damithc) [inline devdocs/staticAnalysis.md]: We already have another `Tool stack` section in another place, and it contains all other tools used. Add this part to that document too?.

**Comment by wkurniawan07** [comment]
@damithc changes made in the last non-merge commit.
The current build has passed static analysis so it's 99% safe to say that it's going to be a passing build.

On a side note, the static analysis run ends at 3 min 30 seconds mark just now. Not sure how it grows with the number of rules, though.

**Comment by wkurniawan07** [comment]
@JosephineKwa hope you don't mind reviewing!

File: src/test/java/teammates/test/cases/ui/browsertests/AllJsTests.java [hunk 3]
```diff
@@ -21,23 +21,23 @@
...
     public void executeJsTests() {
-        String totalCasesXpathQuery = "//span[@class='total']",
-               failedCasesXpathQuery = "//span[@class='failed']";
-        
-        int totalCases = Integer.parseInt(browser.driver
-                .findElement(By.xpath(totalCasesXpathQuery)).getText());
-        int failedCases = Integer.parseInt(browser.driver
-                .findElement(By.xpath(failedCasesXpathQuery)).getText());
+        int totalCases = page.getTotalCases();
+        int failedCases = page.getFailedCases();
```
Comment (JosephineKwa) [inline src/test/java/teammates/test/cases/ui/browsertests/AllJsTests.java]: Is this variable used? Perhaps you could remove it if it isn't :)

Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/AllJsTests.java]: @wkurniawan07 here: yup it is, if the number of failed cases is non-zero this test will fail.

File: src/main/webapp/js/instructorCourseEnrollPage.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ var loadUpFunction = function () {
 
     var ENTER_KEYCODE = 13;
     var enrolTextbox; 
-    if ((enrolTextbox	 = $('#enrollstudents')).length){
+    if ((enrolTextbox     = $('#enrollstudents')).length){
```
Comment (JosephineKwa) [inline src/main/webapp/js/instructorCourseEnrollPage.js]: With regards to the existing style issues, we will be fixing them progressively as new rules are added right?

Comment (wkurniawan07) [inline src/main/webapp/js/instructorCourseEnrollPage.js]: Yes, that's exactly what I had in mind. If I add all the rules at once the diff will easily reach a 4-5 digit number.

File: devdocs/settingUp.md [hunk 5]
```diff
@@ -100,6 +100,12 @@ Important: When a version is specified, please install that version instead of t
    (Make sure you use the `http://localhost:8888/` as the host instead of the one given in the join link)<br>   
    Alternative : Run the test cases, they create several student accounts in the datastore. Use one of them to log in.<br>
 
+## Setting up static analysis tools
```
Comment (JosephineKwa) [inline devdocs/settingUp.md]: Might be good to add this to the table of contents in `settingUp.md` too

Comment (wkurniawan07) [inline devdocs/settingUp.md]: Oops I missed that. It's added now.

**Comment by JosephineKwa** [comment]
The changes look good to me, to the best of my knowledge.

**Comment by JosephineKwa** [comment]
Great work with the PR! This will help to streamline the testing and reviewing process even further :)

**Pull request merged**